src/dns/coredns.rs

```diff
@@ -124,7 +127,8 @@ impl CoreDns {
                         Ok(msg) => {
                             let src_address = msg.addr();
                             let mut dns_resolver = self.resolv_conf.clone();
-                            let sender = sender.clone();
+                            let sender = sender_original.clone();
+                            let sender = sender.with_remote_addr(src_address);
```

Comment:
Any reason this can't be combined into line 130?

Reply:
done.